### PR TITLE
feat: accept docker daemon host via flag

### LINKED ISSUE #170: Support of Docker TCP port to scan local cache images on host machine
Currently, dockle only have a support for `/var/run/docker.sock` to connect Docker Daemon. We @developer-guy noticed that since we rely on `$DOCKER_HOST` variable by setting `tcp://localhost:2375`, dockle ignores the local cache even if image already present on the host machine. Ending up with pulling the image _again_.

Consider following example:
```
scan:
  image: $SCAN_IMAGE
  services:
    - docker:19.03.15-dind
  variables:
    DOCKER_HOST: "tcp://localhost:2375"
    DOCKER_TLS_CERTDIR: ""
  script:
    - docker image pull --quiet $DOCKER_IMAGE
    - dockle --exit-level WARN --exit-code 0 $DOCKER_IMAGE
```

How to reproduce:
1. `$ docker run --privileged -p 12375:2375 -e DOCKER_TLS_CERTDIR="" --rm -it docker:20.10.12-dind`
2. `$ DOCKER_HOST=tcp://localhost:12375 docker image pull busybox`
3. `$ DOCKER_HOST=tcp://localhost:12375 docker image tag busybox foo`
4. 
```
$ DOCKER_HOST=tcp://localhost:12375 docker image ls
REPOSITORY             TAG       IMAGE ID       CREATED        SIZE
busybox                latest    ec3f0931a6e6   10 days ago    1.24MB
foo                    latest    ec3f0931a6e6   10 days ago    1.24MB
```
4. 
```
$ DOCKER_HOST=tcp://localhost:12375 dockle --exit-level WARN --exit-code 0 foo

2022-02-15T17:26:12.772+0300    FATAL   unable to initialize a image struct: failed to initialize source: reading manifest latest in docker.io/library/foo: errors:
denied: requested access to the resource is denied
unauthorized: authentication required
```

If you have necessary access to docker.io, you will get the following error:

```
2022-02-15T12:09:30.033Z	DEBUG	There is no .dockleignore file
2022-02-15T12:09:30.033Z	DEBUG	Fetch latest version from github
2022-02-15T12:09:30.575Z	DEBUG	Start assessments...
2022-02-15T12:09:30.928Z	FATAL	unable to initialize a image struct:
    github.com/goodwithtech/deckoder/extractor/docker.newDockerExtractor
        /home/runner/go/pkg/mod/github.com/goodwithtech/deckoder@v0.0.0-20210909145842-06c0013a5310/extractor/docker/docker.go:73
  - failed to initialize source:
    github.com/goodwithtech/deckoder/extractor/image.NewImage
        /home/runner/go/pkg/mod/github.com/goodwithtech/deckoder@v0.0.0-20210909145842-06c0013a5310/extractor/image/image.go:86
  - manifest unknown: manifest unknown
```

### PR DESCRIPTION
**Pull request by developer-guy**
Fixes #170

Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>
Co-authored-by: Furkan Türkal <furkan.turkal@trendyol.com>

### CODE REVIEW

**Comment by developer-guy** [comment]
kindly ping @tomoyamachi 🙋🏻‍♂️

File: cmd/dockle/main.go [hunk 1]
```diff
@@ -35,6 +36,15 @@ OPTIONS:
   {{end}}{{$option}}{{end}}{{end}}
 `
 	app := cli.NewApp()
+
+	var dockerSockPath string
+	xdgRuntimeDir := os.Getenv("XDG_RUNTIME_DIR")
+	if xdgRuntimeDir != "" {
+		dockerSockPath = fmt.Sprintf("unix://%s/docker.sock", xdgRuntimeDir)
+	} else {
+		dockerSockPath = "unix:///var/run/docker.sock"
```
Comment (tomoyamachi) [inline cmd/dockle/main.go:45]: You can set an empty string if `XDG_RUNTIME_DIR` is not set. So, this line looks unnecessary.

**Review by tomoyamachi — COMMENTED** [review]
I left some comments but overall it looks good to me.

File: cmd/dockle/main.go (hunk 1 above)
Comment (developer-guy) [reply cmd/dockle/main.go:45]: yes, but having it in a variable can help to understand what is going to be the default value of the Docker daemon's socket path, WDYT?

**Comment by tomoyamachi** [comment]
@developer-guy Thank you for your contribution!

**Pull request merged**